orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -36,7 +36,7 @@ class ORTModule(torch.nn.Module):
         debug_options (:obj:`DebugOptions`, optional): debugging options for ORTModule.
     """
 
-    def __init__(self, module, debug_options=None):
+    def __init__(self, module, debug_options=None, provider_configs=None):
```

Comment:
I prefer not to expose these options through the constructor because this config is being used so as to set the execution provider (pytorch device type) in addition to setting the provider config.
So far, we have been using `torch.device` to select the execution provider. But after this change, it becomes ambiguous to the user whether to use `model.to(device)` or to use `model = ORTModule(model, provider_config)`. This might not be the best API from a user perspective. Do we have any other ways to select the provider (maybe use the environment variables so it is hidden)?